File: examples/lh-server/python/03-variables/tasks.py
from typing import Any

from littlehorse.exceptions import LHTaskException


async def fetch_user(user_id: str) -> dict[str, Any]:
    if user_id == "obiwan":
        return {"email": "obiwan@jedi.temple", "title": "Master Kenobi", "age": 37}
    if user_id == "anakin":
        return {"email": "anakin@jedi.temple", "title": "Padawan Anakin", "age": 22}
    raise LHTaskException("user-not-found", "Could not find specified user")


async def send_email(to_address: str, message: str) -> str:
    result = "sent email " + message + " to address " + to_address
    print(result)
    return result

File: examples/lh-server/go/03-variables/tasks.go
package main

import (
	"errors"
	"fmt"
)

type User struct {
	Email string `json:"email"`
	Title string `json:"title"`
	Age   int    `json:"age"`
}

func FetchUser(userId string) (*User, error) {
	switch userId {
	case "obiwan":
		return &User{"obiwan@jedi.temple", "Master Kenobi", 37}, nil
	case "anakin":
		return &User{"anakin@jedi.temple", "Padawan Skywalker (not Master)", 22}, nil
	default:
		return nil, errors.New("user-not-found: Could not find specified user")
	}
}

func SendEmail(toAdress, message string) string {
	result := fmt.Sprintf("sent email %s to %s", message, toAdress)
	fmt.Println(result)
	return result
}
